### ISSUE TITLE
Kinematic 2D Collision Issues

### ISSUE THREAD

**Issue by MajorCooke** [godotengine/godot#20290]
**Godot version:**
3.0.5.stable.official.6a88e22

**OS/device including version:**
Windows 10 Professional

**Issue description:**
Collisions with Kinematic 2D seem to be misbehaving and I've tried about everything that I can to make it work better, yet to no avail. At this point it seems to be a bug.

**Steps to reproduce:**

Download link below.

1. Start the game and let the player rest on the ground, Notice how the player hovers in the air over the first set of blocks.
2. Move to the left and rest on the slopes. It also hovers there. On the right slopes however, the player sinks to the ground normally.
3. The right wall itself, the collision works correctly upon but when you go to the left wall, you can walk right through it up to the leftmost boundary.

**Minimal reproduction project:**
[Download](https://drive.google.com/open?id=1aeaHZh81LwlTBAiiKcn3MxAdBciOqwdd)

**Comment by Ace-Dragon** [comment]
High quality Kinematic body movement for 2D will be far easier to do in Godot 3.1
https://godotengine.org/article/godot-31-will-get-many-improvements-kinematicbody

You can grab daily builds here.
https://hugo.pro/projects/godot-builds/

**Comment by isaacremnant** [comment]
The problem is that your collision shapes are shifted (Debug > Visible collision shapes):
![prob](https://user-images.githubusercontent.com/35355161/42975532-96c76686-8b8a-11e8-8cca-cfbc41025faf.png)

I'm not sure if this is something you did wrong when creating your tiles, but there's a mismatch between the .tscn and .tres:
![prob2](https://user-images.githubusercontent.com/35355161/42975617-0e7b86d0-8b8b-11e8-8467-52fb59596f98.png)
And here's the relevant part in the tileset's .tres:
```
[sub_resource type="ConvexPolygonShape2D" id=6]

custom_solver_bias = 0.0
points = PoolVector2Array( -8, 8, 8, -8, 8, 8 )
```

As you can see, your collision shapes are offset by (-8, -8). You can fix your problem by manually shifting all those ConvexPolygonShape2D by (+8, +8).

**That said**, I did get this issue a few times... my tiles got somehow shifted in a similar fashion during creation and I didn't think much of it. I think it's worth investigating this a bit further...

**Comment by MajorCooke** [comment]
@isaacremnant I'll take a look into it, but thanks for pointing that out. I did not try to shift the collision bounds at all here too so yes, I do think this should be investigated too.

**Comment by MajorCooke** [comment]
@isaacremnant I'll take a look into it, but thanks for pointing that out. I did not try to shift the collision bounds at all here too so yes, I do think this should be investigated too.

It's strange because setting it to 0,0 [does this.](https://puu.sh/AZIhB/064c2313c8.png)

(Side question, how did you get that tab at the bottom to show collisions? Never seen that before and I'm still quite new to this.)

**Comment by MajorCooke** [comment]
@isaacremnant That did the trick. By setting the **2DStaticBody** offsets from (8,8) to (0,0) and then manually offsetting the collision boxes indeed to (8,8) fixed it.

Thank you.

I'm leaving this open though because I think it should be investigated too.

**Comment by bojidar-bg** [comment]
I can't confirm this issue with the sample project given by the OP on 3.1, assuming fixed by #19343.

If someone can confirm this on **3.1**, this issue might be reopened.
